SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialLoadBalanceEnforcer.cpp

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "EngineClasses/SpatialLoadBalanceEnforcer.h"
+#include "EngineClasses/SpatialVirtualWorkerTranslator.h"
+#include "EngineClasses/SpatialNetDriver.h"
+#include "Interop/Connection/SpatialWorkerConnection.h"
+#include "Interop/SpatialSender.h"
+#include "Interop/SpatialStaticComponentView.h"
+#include "Schema/AuthorityIntent.h"
+#include "SpatialCommonTypes.h"
+
+DEFINE_LOG_CATEGORY(LogSpatialLoadBalanceEnforcer);
+
+using namespace SpatialGDK;
+
+USpatialLoadBalanceEnforcer::USpatialLoadBalanceEnforcer(const FObjectInitializer& ObjectInitializer)
+	: Super(ObjectInitializer)
+	, NetDriver(nullptr)
+	, Sender(nullptr)
+	, VirtualWorkerTranslator(nullptr)
+{
+}
+
+void USpatialLoadBalanceEnforcer::Init(USpatialNetDriver* InNetDriver,
+	USpatialSender* InUpatialSender,
+	SpatialVirtualWorkerTranslator* InVirtualWorkerTranslator)
+{
+	NetDriver = InNetDriver;
+	Sender = InUpatialSender;
+	VirtualWorkerTranslator = InVirtualWorkerTranslator;
+}
+
+void USpatialLoadBalanceEnforcer::Tick()
+{
+	ProcessQueuedAclAssignmentRequests();
+}
+
+void USpatialLoadBalanceEnforcer::OnComponentUpdated(const Worker_ComponentUpdateOp& Op)
+{
+	check(NetDriver != nullptr)
+	check(NetDriver->StaticComponentView != nullptr)
+	if (Op.update.component_id == SpatialConstants::AUTHORITY_INTENT_COMPONENT_ID &&
+		NetDriver->StaticComponentView->GetAuthority(Op.entity_id, SpatialConstants::ENTITY_ACL_COMPONENT_ID) == WORKER_AUTHORITY_AUTHORITATIVE)
+	{
+		QueueAclAssignmentRequest(Op.entity_id);
+	}
+}
+
+void USpatialLoadBalanceEnforcer::AuthorityChanged(const Worker_AuthorityChangeOp& AuthOp)
+{
+	if (AuthOp.component_id == SpatialConstants::ENTITY_ACL_COMPONENT_ID &&
+		AuthOp.authority == WORKER_AUTHORITY_AUTHORITATIVE)
+	{
+		// We have gained authority over the ACL component for some entity
+		// If we have authority, the responsibility here is to set the EntityACL write auth to match the worker requested via the virtual worker component
```

Comment:
This is a confusing sentence to read. Is the first half redundant?

Reply:
This all kinda of seems redundant.